fusestore/store/src/meta_service/raft_types.rs

```diff
@@ -21,3 +25,26 @@ impl SledOrderedSerde for u64 {
         BigEndian::read_u64(buf)
     }
 }
+
+/// For LogId to be able to stored in sled::Tree as a key.
+impl SledOrderedSerde for String {
+    fn ser(&self) -> Result<IVec, ErrorCode> {
+        Ok(IVec::from(self.as_str()))
+    }
+
+    fn de<V: AsRef<[u8]>>(v: V) -> Result<Self, ErrorCode>
+    where Self: Sized {
+        Ok(String::from_utf8(v.as_ref().to_vec())?)
+    }
+
+    fn order_preserved_serialize(&self, _buf: &mut [u8]) {
+        todo!()
+    }
+
+    fn order_preserved_deserialize(_buf: &[u8]) -> Self {
+        todo!()
+    }
```

Comment:
- FYI: There is a so-called "Memory Comparable Encoding" in TiKV

https://pingcap.com/blog-cn/tikv-source-code-reading-13/ 

the relevant doc is a little bit oversimplified  (for me), taking a look at the  code may be helpful

https://github.com/tikv/tikv/blob/6708ce171792df02e1e90e1fe1e67e424d1586c8/components/codec/src/byte.rs#L64

- Is it the right time (or in the near future) to change the type of KEY (no longer stick to &str)?
  User-defined key types and order-preserving key encoding are easier to be implemented in binary format.

Reply:
Sorry for the confusion. This is not really a `todo`. This function is not called. Please ignore these two todo.
Another refactoring PR is on the way!


> * Is it the right time (or in the near future) to change the type of KEY (no longer stick to &str)?
>   User-defined key types and order-preserving key encoding are easier to be implemented in binary format.

Have you ever run into a situation where you have to use bytes instead of str as the key?

Reply:
got it